mcos/optimizer.py

```diff
@@ -7,6 +7,9 @@
 from sklearn.cluster import KMeans
 from sklearn.metrics import silhouette_samples
 from mcos.covariance_transformer import cov_to_corr
+from typing import List
+import scipy.cluster.hierarchy as sch
```

Comment:
Could we move these two external imports above `from mcos.covariance_transformer import cov_to_corr`?